source/common/http/http2/metadata_interface.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <unordered_map>
+
+namespace Envoy {
+namespace Http {
+namespace Http2 {
+
+/**
+ * Please refer to #2394 for more info about Envoy METADATA.
+ * TODO(soya3129): refer to Envoy document.
+ *
+ * METADATA frame is an Envoy specific extension for the standard HTTP/2 protocol. METADATA frame
+ * can be used to communicate extra information related to a session or a stream. HTTP/2 protocol
```

Comment:
Can you clarify that metadata (as I understand it) is basically a header map that can appear anywhere within the stream as opposed to trailers? Is that correct?

Reply:
That's correct! Added that info to the comments.